components/_Progress-Bar/progress-bar-initialization.js

```diff
@@ -0,0 +1,111 @@
+    if(ProgressBarDefinition) {
+        progressBar.init("ProgressBarDefinition", 5, 4); 
+    }
+
+    //Progress Bar Features scripts------------------    
+    const progressBarFeatures1El = document.getElementById("ProgressBarFeatures1");
+    if(progressBarFeatures1El) {
+        //Initialize the progress bar with 5 steps starting form the default value 0.
+        const progressBarFeatures1 = Object.create(window.msdsProgressBar);
+        progressBarFeatures1.init("ProgressBarFeatures1", 5, 0);
+    }
+        
+    const progressBarFeatures2El = document.getElementById("ProgressBarFeatures2");
+        if(progressBarFeatures2El) {
+        //Initialize the progress bar with 5 steps starting form the 1.
+        const progressBarFeatures2 = Object.create(window.msdsProgressBar);
+        progressBarFeatures2.startFromOne = true; //Set "StartFromOne" variable to true.
+        progressBarFeatures2.init("ProgressBarFeatures2", 5, 0); //Initialize the progress bar with 5 steps.
+    }
+
+    const progressBarFeatures3El = document.getElementById("ProgressBarFeatures3");
+    if(progressBarFeatures3El) {
+        //Set the progress bar to a specific step.
+        const progressBarFeatures3 = Object.create(window.msdsProgressBar);
+        progressBarFeatures3.init("ProgressBarFeatures3", 5, 3); //Initialize the progress bar with 5 steps.
+    }
+
+    //Progress bar Size scripts------------------
+    const progressBarSize1El = document.getElementById('ProgressBarSize1');
+    if(progressBarSize1El) {
+        const progressBarSize1 = Object.create(window.msdsProgressBar);
+        progressBarSize1.init('ProgressBarSize1', 5, 2); //Initialize the progress bar with 5 steps.
+    }    
+            
+    const progressBarSize2El = document.getElementById('ProgressBarSize2');
+    if(progressBarSize2El) {
+        //Set the progress bar to a specific step by default.
+        const progressBarSize2 = Object.create(window.msdsProgressBar);      
+        progressBarSize2.init('ProgressBarSize2', 5, 3); //Initialize the progress bar with 5 steps.
+    }
+     
+    
+    //Light Theme scripts------------------       
+    //Initialize the progress bar with 12 steps.        
+    const lightThemeProgressBarEl = document.getElementById("LightThemeProgressBar");
+    if(lightThemeProgressBarEl) {
+        const lightThemeProgressBar = Object.create(window.msdsProgressBar);
+
+        let lightThemeBaseNumberElement = document.querySelectorAll('#LightThemeBaseNumber')[0]    
+        lightThemeBaseNumberElement.value = 12;
+
+        lightThemeProgressBar.init("LightThemeProgressBar", lightThemeBaseNumberElement.value, 0);  
+        
+        document.querySelector('#LightThemePrevStep').addEventListener('click', event => {            
+            lightThemeProgressBar.prev();
+        })
+    
+        document.querySelector('#LightThemeNextStep').addEventListener('click', event => {
```

Comment:
I would suggest extract these out of this method and bind them somewhere en a bindEvents() method

Reply:
and then put the logic into its own method which are then called via the bindEvent eventlistener. That way you automatically reduce the size of this method by alot. This should be done to all event listensers in my opinion

Reply:
again this file is for demo only and I did not want to spend hours and make it as pretty. this event binding is just for us to see that they are there. what you suggest would be done in a specific js file inside our block definition but not here. 
I see your point and might come up with a better solution how to initialize prev, next and pagination.

Reply:
I totally understand you, but as we might need to maintain this in the future, the clarity of this code is just as important than the rest of it, in my opinion :)